lib/checkersreport.cpp

```diff
@@ -280,3 +280,89 @@ std::string CheckersReport::getReport(const std::string& criticalErrors) const
 
     return fout.str();
 }
+
+std::string CheckersReport::getXmlReport(const std::string& criticalErrors) const
+{
+    std::string ret;
+
+    ret += "    <critical-errors>";
+    if (!criticalErrors.empty())
+        ret += criticalErrors + "\n";
+    ret += "</critical-errors>\n";
+    ret += "    <checkers-report>\n";
+
+    const bool cppcheckPremium = isCppcheckPremium(mSettings);
+
+    auto reportSection = [&ret, cppcheckPremium]
+                             (const std::string& title,
+                             const Settings& settings,
+                             const std::set<std::string>& activeCheckers,
+                             const std::map<std::string, std::string>& premiumCheckers,
+                             const std::string& substring) {
+        ret += "        <" + title + ">";
+        if (!cppcheckPremium) {
+            ret += "Not available, Cppcheck Premium is not used</" + title + ">\n";
+            return;
+        }
+        ret += "\n";
+        for (const auto& checkReq: premiumCheckers) {
+            const std::string& checker = checkReq.first;
+            if (checker.find(substring) == std::string::npos)
+                continue;
+            bool active = cppcheckPremium && activeCheckers.count(checker) > 0;
+            if (substring == "::") {
+                if (checkReq.second == "warning")
+                    active &= settings.severity.isEnabled(Severity::warning);
+                else if (checkReq.second == "style")
+                    active &= settings.severity.isEnabled(Severity::style);
+                else if (checkReq.second == "portability")
+                    active &= settings.severity.isEnabled(Severity::portability);
+                else if (!checkReq.second.empty())
+                    active = false; // FIXME: handle req
+            }
+            ret += "            <checker active=\"" + static_cast<std::string>(active ? "Yes" : "No") + "\" id=\"" + checker + "\"";
```

Comment:
`static_cast<std::string>(` .. casting to a std::string looks convoluted to me. Just construct the std::string explicitly.